### PR TITLE
fix: converting string to string list for TYPE_STRING_ARRAY attribute kind

### PR DESCRIPTION
**Pull request by tim-mwangi**

### CODE REVIEW

File: gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/StringToAttributeKindConverter.java [hunk 1]
```diff
@@ -14,7 +15,8 @@
 
 public class StringToAttributeKindConverter extends ToAttributeKindConverter<String> {
   private static final Logger log = LoggerFactory.getLogger(StringToAttributeKindConverter.class);
```
Comment (buchi-busireddy) [inline gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/StringToAttributeKindConverter.java:17]: Nit. `s/log/LOGGER/`

File: gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/StringToAttributeKindConverter.java [hunk 2]
```diff
@@ -69,20 +71,38 @@ public Value doConvert(String value, AttributeKind attributeKind, Value.Builder
...
+  private List<String> convertToArray(String jsonString) {
+    if (StringUtils.isEmpty(jsonString)) {
+      return List.of();
+    }
+
+    try {
+      return objectMapper.readValue(jsonString, LIST_TYPE_REFERENCE);
+    } catch (IOException e) {
+      log.warn("Unable to read List JSON String data from: {}. With error:", jsonString, e);
+      log.warn("Setting data as empty list instead");
```
Comment (buchi-busireddy) [inline gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/StringToAttributeKindConverter.java:93]: Do we need the second warning? it should be implicit or merge it with the above line itself.

File: gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/StringToAttributeKindConverter.java (hunk 1 above)
Comment (tim-mwangi) [reply gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/StringToAttributeKindConverter.java:17]: done

File: gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/StringToAttributeKindConverter.java (hunk 2 above)
Comment (tim-mwangi) [reply gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/StringToAttributeKindConverter.java:93]: Will merge into the other log message.

**Review by buchi-busireddy — APPROVED** [review]

**Pull request merged**